kore/test/Test/Kore/Builtin/KEqual.hs

```diff
@@ -55,30 +55,6 @@ test_KEqual =
         actual <- evaluate original
         assertEqual' "" expect actual
 
-    , testCaseWithSMT "kseq(x, dotk) equals kseq(x, dotk)" $ do
```

Comment:
Why are these tests removed?

Reply:
I initially removed them because they were dependent on `==K` simplification and that was handled by applying the frontend rules. It looks like they fail even with the internal implementation, I'm looking into what might be wrong.

Reply:
Update: the term needed to be an `and true ...` or an `equals true ...` in order for the internal simplification to work; I'll update the tests.